src/lib/post.ts

```diff
@@ -264,4 +262,180 @@ const feed = async (req: IGetUserAuthRequest, res: Response) => {
+			return res.status(400).json({
+				message: 'Post already liked',
+			});
+		}
+		post.likes?.push(user._id);
+		const savedPost = await post.save();
+		logger.info('Post liked');
+		return res.status(200).json({
+			message: 'Post liked',
+			post: savedPost,
+		});
+	} catch (error: any) {
+		logger.error(error.message);
+		return res.status(400).json({
+			message: error.message,
+		});
+	}
+};
+
+/**
+ * @desc Unlike a post
+ * @route PUT /post/unlike/:id
+ * @access Authenticated
+ * @Param postId
+ * @access Moderator can't unlike post
+ * @access User can't unlike post if he has no friends
+ */
+const unlike = async (req: IGetUserAuthRequest, res: Response) => {
+	const { postId } = req.params;
+	const { user } = req;
+	try {
+		const post = await Post.findById(postId);
+		if (!post) {
+			logger.info('Post not found');
+			return res.status(400).json({
+				message: 'Post not found',
+			});
+		}
+		if (!post.likes?.includes(user._id)) {
+			logger.info('Post not liked');
+			return res.status(400).json({
+				message: 'Post not liked',
+			});
+		}
+		post.likes = post.likes?.filter((like) => like !== user._id);
+		const savedPost = await post.save();
+		logger.info('Post unliked');
+		return res.status(200).json({
+			message: 'Post unliked',
+			post: savedPost,
+		});
+	} catch (error: any) {
+		logger.error(error.message);
+		return res.status(400).json({
+			message: error.message,
+		});
+	}
+};
+
+/**
+ * @desc Get all comments with replies in graph format
+ * @route GET /post/:id/comments
+ * @access Authenticated
+ * @Param id
+ * @Query page, limit, desc
+ * @access Moderator can't get comments
+ * @access User can't get comments if post is deleted
+ * @access User can't get comments if post is not his and he is not following the user
+ */
+const getComments = async (req: IGetUserAuthRequest, res: Response) => {
+	const { id } = req.params;
+	const { user } = req;
+	try {
+		const post = await Post.findById(id);
+		if (!post) {
+			logger.info('Post not found');
+			return res.status(400).json({
+				message: 'Post not found',
+			});
+		}
+		if (post.deleted) {
+			logger.info('Post is deleted');
+			return res.status(400).json({
+				message: 'Post is deleted',
+			});
+		}
+		if (
+			post.createdBy.toString() !== user._id.toString() &&
+			!user.followedUsers.includes(post.createdBy)
+		) {
+			logger.info('Unauthorized');
+			return res.status(401).json({
+				message: 'Unauthorized',
+			});
+		}
+		const comments = await Comment.find({ postId: id })
+			.where('replyTo')
+			.equals(null);
+
+		// const commentsWithReplies = await Promise.all(
+		// 	comments.map(async (comment) => {
+		// 		const replies = await Comment.find({ parentCommentId: comment._id });
+		// 		return {
+		// 			...comment.toObject(),
+		// 			replies,
+		// 		};
+		// 	})
+		// );
+
+		//loop through comments and add replies to each comment and add replies to each reply and so on
+		const commentsWithReplies = await Promise.all(
```

Comment:
1. Move this part to a separate function so it could be reused
2. Why Promise.all is needed?